openmdao/docs/theory_manual/total_derivs/setup_linear_solvers.rst

```diff
@@ -0,0 +1,312 @@
+Jacobian matrix up front, and then hold onto that and re-use it throughout the run.
+This has several computational advantages, but the major one is that it helps to reduce framework
+overhead for models with deep system hierarchies and large numbers of variables.
+
+You should strongly consider using an assembled Jacobian if all the components in your model provide
+derivatives using the :ref:`compute_partials <comp-type-2-explicitcomp>` or
+:ref:`linearize <comp-type-3-implicitcomp>` methods.
+These methods are explicitly computing the elements of that Jacobian matrix, and so it makes sense
+to collect them into an actual matrix memory representation.
+
+Additionally, if your model has a very large hierarchy (i.e. many levels, many components,
+many variables) then an assembled Jacobian will likely offer a significant performance advantage.
+The reason that large models benefit is that without the assembled Jacobian, OpenMDAO must
+recursively loop over each level of the hierarchy, each component, and each variable in order to
+compute Jacobian-vector products.
+That triple for-loop is rather expensive, and it's much more efficient to collect the Jacobian in a
+single chunk of memory if possible.
+So even if you are using an iterative linear solver, such as :ref:`ScipyKrylov<scipyiterativesolver>`
+or :ref:`PetscKrylov<petscKrylov>`, an assembled Jacobian is generally more efficient.
+
+.. note::
+    If you want to see how to add an assembled Jacobian to your model, check out
+    :ref:`this feature doc <feature_assembled_jacobian>`.
+
+
+Sparse Assembled Jacobian
+**************************
+In the majority of cases, if an assembled Jacobian is appropriate for your model then you want to
+use the :ref:`CSCJacobian<openmdao.jacobians.assembled_jacobian.py>`.
+The :ref:`CSCJacobian<openmdao.jacobians.assembled_jacobian.py>` only allocates memory for the
+non-zero partial derivatives.
+How does OpenMDAO know which partials are non-zero?
+The authors of the components in your model declared them using either a
+:ref:`dense<feature_specify_partials>` or :ref:`sparse<feature_sparse_partials>` form of :code:`declare_partials`.
+
+Even if all of your components declared their partial derivatives as dense (or if they are all
+scalar variables and specifying sparsity doesn't have meaning), at the group level there is still a
+sparsity pattern to be taken advantage of.
+This sparsity arises from the way components are connected to one another, because unless there is
+a connection present, there is no need to allocate space associated with that portion of the Jacobian.
+We can see this clearly by looking at a collapsed form of the :math:`N^2` diagram with just the outputs shown.
+There are 7 scalar outputs, so we have a :math:`7 \times 7` partial derivative Jacobian.
+Out of the possible 49 matrix entries, only 18 are actually non-zero.
+That makes it 37% sparse.
```

Comment:
This seems backwards.  Isn't it actually 37% dense (or 63% sparse)?  And in a real problem with thousands of variables, with the sparsity be above 90%?